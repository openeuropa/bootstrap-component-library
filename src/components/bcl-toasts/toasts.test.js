import { merge, renderTwigFileAsNode } from "@openeuropa/bcl-test-utils";
import demoSingle from "@openeuropa/bcl-data-toasts/data";
import demoMultiple from "@openeuropa/bcl-data-toasts/data-multiple";

const template = "@oe-bcl/bcl-toasts/toasts.html.twig";
const render = (params) => renderTwigFileAsNode(template, params);

describe("OE - Toasts", () => {
  test(`renders correctly with a single toast`, () => {
    expect.assertions(1);

    return expect(render(demoSingle)).resolves.toMatchSnapshot();
  });

  test(`render correctly with multiple toasts`, () => {
    expect.assertions(1);

    return expect(render(demoMultiple)).resolves.toMatchSnapshot();
  });
});
