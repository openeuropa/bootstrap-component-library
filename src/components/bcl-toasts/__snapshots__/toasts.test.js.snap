// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OE - Toasts render correctly with body wrapper classes 1`] = `
<jest>
  <div
    aria-atomic="true"
    aria-live="assertive"
    class="toast"
    role="alert"
  >
    <div
      class="wrapper-class another-wrapper-class"
    >
      <div
        class="toast-body"
      >
        Hello, world! This is a toast message.
            
        <div
          class="mt-2 pt-2 border-top"
        >
          <button
            class="btn btn-primary btn-sm me-1"
            type="button"
          >
            Take action
          </button>
          <button
            class="btn btn-secondary btn-sm"
            data-bs-dismiss="toast"
            type="button"
          >
            Close
          </button>
        </div>
      </div>
    </div>
  </div>
</jest>
`;

exports[`OE - Toasts render correctly with multiple toasts 1`] = `
<jest>
  <div
    aria-atomic="true"
    aria-live="polite"
    class="position-relative"
  >
    <div
      class="toast-container position-absolute top-0 end-0 p-3"
    >
      <div
        aria-atomic="true"
        aria-live="assertive"
        class="toast"
        role="alert"
      >
        <div
          class="toast-header"
        >
          <strong
            class="me-auto"
          >
            Bootstrap
          </strong>
          <small
            class="text-muted"
          >
            just now
          </small>
          <button
            aria-label="close"
            class="btn-close"
            data-bs-dismiss="toast"
            type="button"
          />
        </div>
        <div
          class="toast-body"
        >
          See? Just like this.
        </div>
      </div>
      <div
        aria-atomic="true"
        aria-live="assertive"
        class="test-class toast"
        role="alert"
      >
        <div
          class="toast-header"
        >
          <strong
            class="me-auto"
          >
            Bootstrap
          </strong>
          <small
            class="text-muted"
          >
            2 seconds ago
          </small>
          <button
            aria-label="close"
            class="btn-close"
            data-bs-dismiss="toast"
            type="button"
          />
        </div>
        <div
          class="toast-body"
        >
          Heads up, toasts will stack automatically.
        </div>
      </div>
    </div>
  </div>
</jest>
`;

exports[`OE - Toasts renders correctly with a single toast 1`] = `
<jest>
  <div
    aria-atomic="true"
    aria-live="assertive"
    class="toast"
    role="alert"
  >
    <div
      class="toast-body"
    >
      Hello, world! This is a toast message.
            
      <div
        class="mt-2 pt-2 border-top"
      >
        <button
          class="btn btn-primary btn-sm me-1"
          type="button"
        >
          Take action
        </button>
        <button
          class="btn btn-secondary btn-sm"
          data-bs-dismiss="toast"
          type="button"
        >
          Close
        </button>
      </div>
    </div>
  </div>
</jest>
`;
